### DISCUSSION TITLE
Failing  to run `npx prisma generate`

### DISCUSSION THREAD

**Discussion by devdaim6** [prisma/prisma#23165] [Q&A]
### Bug description

Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Error: Get DMMF: NodeAPIQueryEngineLibrary.dmmf is not a function

### How to reproduce

I dont know :(

### Prisma information

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

generator zod {
  provider              = "zod-prisma"
  output                = "./zod" 
  relationModel         = true 
  modelCase             = "camelCase" 
  modelSuffix           = "Schema" 
  useDecimalJs          = true 
  prismaJsonNullability = true 
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}


model Account {
  id                 String  @id @default(cuid())
  userId             String
  type               String
  provider           String
  providerAccountId  String
  refresh_token      String?  @db.Text
  access_token       String?  @db.Text
  expires_at         Int?
  token_type         String?
  scope              String?
  id_token           String?  @db.Text
  session_state      String?

  refresh_token_expires_in Int?

  user User @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
}

model Session {
  id           String   @id @default(cuid())
  sessionToken String   @unique
  userId       String
  expires      DateTime
  user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
}

model User {
  id            String    @id @default(cuid())
  name          String?
  email         String?   @unique
  emailVerified DateTime?
  image         String?
  accounts      Account[]
  sessions      Session[]
}

model VerificationToken {
  identifier String
  token      String   @unique
  expires    DateTime

  @@unique([identifier, token])
}
```

```ts
import { PrismaClient } from "@prisma/client";

declare global {
  // allow global `var` declarations
  // eslint-disable-next-line no-var
  var db: PrismaClient | undefined;
}

export const db =
  global.db ||
  new PrismaClient({
    log: ["query"],
  });

if (process.env.NODE_ENV !== "production") global.db = db;


```

### Environment & setup

- OS: Windows 11
- Database: PostgreSQL
- Node.js version: 21.5.0

### Prisma Version

```prisma
Prisma CLI Version : 3.15.2
```

**Comment by devdaim6** [comment]
![prisma version.png](https://github.com/prisma/prisma/assets/123406647/cf049b1a-d3bd-403f-9ed4-019ddde07f4b)

got the following output of the command you suggested.
Appreciated if anybody could help me with this issue.
Thanks 😊

**Reply by nurul3101** [reply]
@devdaim6 Can you please use the same version in `prisma` as well as `@prisma/client` npm package?

Try using `5.9.1` for both of these packages. This should hopefully fix the issue for you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
